frontend/src/scene/payment/state.js

```diff
@@ -0,0 +1,35 @@
+import { observable, decorate, action } from 'mobx';
+import { getPaymentDetails } from '../../apis';
+
+class PaymentFormState {
+    order_number = '';
+    submitError = true;
+    submitSuccess = false;
+
+    constructor(userStore) {
+        this.userStore = userStore;
+        this.balance = this.userStore.balance;
```

Comment:
you can directly access balance anyway through userStore, this is a duplication in code

Reply:
fixed